base/essentials.jl

```diff
@@ -165,6 +165,14 @@ function convert end
 
 convert(::Type{Any}, @nospecialize(x)) = x
 convert(::Type{T}, x::T) where {T} = x
+
+unwrap(x) = x
+
+function convert(::Type{T}, x) where {T}
+    y = unwrap(x)
+    return y === x ? x : convert(T, y)
```

Comment:
MethodError if `y === x`?

Reply:
Yeah, I wasn't quite sure on how this should interact (replace?) `convert(Any, x)`, so I just simulated convert to Any

Reply:
But `convert(T, x)` is supposed to return something of type `T`. Returning `x` works for `Any` but not all `T`.